pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,124 @@ package sdkstats
+				} else if err != nil {
+					return nil, err
+				}
+				repoList = append(repoList, *repoDetails)
 			}
-			repoList = append(repoList, *repoDetails)
-
 		}
 	}
-
 	return repoList, nil
 }
 
-//Search returns Github API results for Code Search, for a given Query string
-func Search(client *github.Client, q string) ([]*github.CodeSearchResult, error) {
-
-	var searchOp []*github.CodeSearchResult
-	ctx := context.Background()
+// search returns Github API results for Code Search, for a given Query string
+func (c Client) search(ctx context.Context, q string) ([]*github.CodeSearchResult, error) {
+	var searchop []*github.CodeSearchResult
 
 	opts := &github.SearchOptions{TextMatch: true,
 		ListOptions: github.ListOptions{
 			PerPage: 100,
 		},
 	}
+	//Below Logic enables to collate Search results from all pages returned from API call.
 	for {
-		op, resp, err := client.Search.Code(ctx, q, opts)
+		op, resp, err := c.Client.Search.Code(ctx, q, opts)
 		if err != nil {
 			return nil, err
 		}
-		searchOp = append(searchOp, op)
+		searchop = append(searchop, op)
 		if resp.NextPage == 0 {
 			break
 		}
 		opts.Page = resp.NextPage
 	}
-	return searchOp, nil
-
+	return searchop, nil
 }
 
-//GetRepoDetails Returns map[string][string] with Stars,CreatedAt,PushedAt, and TotalCommits details for a given onwer, and repo
-func GetRepoDetails(client *github.Client, owner string, name string) map[string]string {
-
-	repoMap := map[string]string{}
+// getRepoDetails Returns []RepoMetaData with Stars,CreatedAt,PushedAt, and TotalCommits details for a given owner, and repo.
+func (c Client) getRepoDetails(ctx context.Context, codeResults github.CodeResult, rq RepoMetadataQuery) (*RepoMetadata, error) {
 
-	ctx := context.Background()
+	repoOwner := codeResults.GetRepository().GetOwner().GetLogin()
+	repoName := codeResults.GetRepository().GetName()
 
-	repos, _, err := client.Repositories.Get(ctx, owner, name)
+	version, err := rq.VersionParser.ParseVersion(codeResults)
 	if err != nil {
-		fmt.Println(err)
+		version = "N/A"
 	}
-	commits, _, err := client.Repositories.ListContributorsStats(ctx, owner, name)
-	var total int
-	for _, r := range commits {
-		total = total + r.GetTotal()
-	}
-	repoMap["Stars"] = strconv.Itoa(repos.GetStargazersCount())
-	repoMap["CreatedAt"] = repos.GetCreatedAt().String()
-	repoMap["PushedAt"] = repos.GetPushedAt().String()
-	repoMap["TotalCommits"] = strconv.Itoa(total)
-	return repoMap
-
-}
 
-func getversion(codeResults github.CodeResult, projectType string) int {
-
-	var searchQ string
-	var SDKversion int
-
-	if projectType == "helm" {
-		searchQ = "quay.io/operator-framework/helm-operator:v0."
+	repo, _, err := c.Client.Repositories.Get(ctx, repoOwner, repoName)
+	if err != nil {
+		return nil, err
 	}
-	if projectType == "ansible" {
-		searchQ = "quay.io/operator-framework/ansible-operator:v0."
+
+	totalCommits := 0
+	commits, _, err := c.Client.Repositories.ListContributorsStats(ctx, repoOwner, repoName)
```

Comment:
It can get directly. Call : curl -i https://api.github.com/repos/operator-framework/operator-sdk parse the JSON and the result will be there. 

```
$ curl -i https://api.github.com/repos/operator-framework/operator-sdk
...
  "stargazers_count": 2647,
  "watchers_count": 2647,
..
  "forks_count": 701,
 ..
  "open_issues_count": 131,
..
  "forks": 701,
  "open_issues": 131,
  "watchers": 2647,
..
  "network_count": 701,
  "subscribers_count": 92
}
```